rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -1064,12 +1068,24 @@ def _maybe_decode_erc20_721_transfer(
         else:
             found_token = token
 
-        transfer = self.base.decode_erc20_721_transfer(
+        if (  # detect if the token is impersonating the base asset currency
```

Comment:
why not do this in `get_or_create_evm_token()`?

minor thing: is it an overkill if you make this into a function in `spam_assets.py` and use that,